### PR TITLE
feat(studio): DD schema list in builder

### PR DESCRIPTION
**Pull request by steramae-nvidia**
Screenshot 2026-07-22 at 11 47 59 AM

Signed-off-by: Sean Teramae <steramae@nvidia.com>

## Summary by CodeRabbit

* **New Features**
  * Added a “list” view to the Data Designer with selectable, deletable schema rows and concise per-column descriptions.
  * Introduced a builder view toggle (list vs. canvas) for quicker column inspection.
* **Bug Fixes**
  * Improved column ordering by dependency relationships, including safer behavior for cyclic dependencies.
* **Tests**
  * Added tests for column sorting and description rendering, including edge cases for judge and generator details.

### CODE REVIEW

**Comment by steramae-nvidia** [comment]
This change is part of the following stack:

- #847 ◀
    - #867

<sub>Change managed by [git-spice](https://abhinav.github.io/git-spice/).</sub>

File: web/packages/studio/src/routes/DataDesignerJobBuildRoute/SchemaRow.tsx [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+            </Flex>
+          ) : null}
+        </Stack>
+      </button>
+
+      <button
+        type="button"
+        onClick={onDelete}
+        aria-label={`Delete ${column.name || option.label}`}
+        className="flex shrink-0 items-center border-l border-base px-density-md text-secondary opacity-0 transition-opacity hover:text-(--text-color-feedback-danger) focus-visible:opacity-100 focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-inset focus-visible:ring-(--color-brand,#76b900) group-hover:opacity-100 cursor-pointer"
```
Comment (nakolean) [inline web/packages/studio/src/routes/DataDesignerJobBuildRoute/SchemaRow.tsx]: **Nit**: This is a lot of classNames for a native button, could we not use a KUI Button and style it here?

**Review by nakolean — APPROVED** [review]
just one nitpick, looks good to me

File: web/packages/studio/src/routes/DataDesignerJobBuildRoute/SchemaRow.tsx (hunk 1 above)
Comment (steramae-nvidia) [reply web/packages/studio/src/routes/DataDesignerJobBuildRoute/SchemaRow.tsx]: true let me give it a shot
